### PR TITLE
Added Phalcon DI Container Adapter

### PR DESCRIPTION
**Pull request by theluk**
One issue in this commit is, that as Phalcon cannot be just installed using composer, as it needs to be compiled, I made the tests skippable if phalcon is not present using the following comment

    /**
     * @requires extension phalcon
     */

That produces three `S` (skipped) in the command line test runner. I have not found a nicer method.

### CODE REVIEW

File: src/Adapter/PhalconDIContainerAdapter.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        $this->container = $container;
+    }
+
+    public function get($id)
+    {
+        try {
+            return $this->container->get($id);
+        } catch(PhalconDIException $prev) {
+            throw AcclimateNotFoundException::fromPrevious($id);
```
Comment (jeremeamia) [inline src/Adapter/PhalconDIContainerAdapter.php]: Missing `$prev` in `fromPrevious()`.

File: src/Adapter/PhalconDIContainerAdapter.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    public function __construct(PhalconDI $container)
+    {
+        $this->container = $container;
+    }
+
+    public function get($id)
+    {
+        try {
+            return $this->container->get($id);
+        } catch(PhalconDIException $prev) {
```
Comment (jeremeamia) [inline src/Adapter/PhalconDIContainerAdapter.php]: Need space between `catch` and `(`.

File: tests/Adapter/PhalconDIContainerAdapterTest.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace Acclimate\Container\Test\Adapter;
+
+use Acclimate\Container\Adapter\PhalconDIContainerAdapter;
+use Phalcon\DI as PhalconDI;
+
+/**
+ * @requires extension phalcon
+ */
+class PhalconDIContainerAdapterTest extends AbstractContainerAdapterTest {
```
Comment (jeremeamia) [inline tests/Adapter/PhalconDIContainerAdapterTest.php]: `{` on next line, please.

File: tests/Adapter/PhalconDIContainerAdapterTest.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        $container = new PhalconDI();
+        $container->setShared('array_iterator', new \ArrayIterator(range(1, 5)));
+        $container->set('error', function() {
+            throw new \RuntimeException;
+        });
+
+        return new PhalconDIContainerAdapter($container);
+    }
+
```
Comment (jeremeamia) [inline tests/Adapter/PhalconDIContainerAdapterTest.php]: No blank lines at the beginning or end of the class.

File: src/Adapter/PhalconDIContainerAdapter.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Acclimate\Container\Adapter;
+
+use Interop\Container\ContainerInterface as AcclimateContainerInterface;
+use Acclimate\Container\Exception\NotFoundException as AcclimateNotFoundException;
+use Acclimate\Container\Exception\ContainerException as AcclimateContainerException;
+
```
Comment (jeremeamia) [inline src/Adapter/PhalconDIContainerAdapter.php]: No need for the empty line here.

**Comment by jeremeamia** [comment]
There are a few style things and one bug that I'd like you to fix up, but otherwise, it looks good. Thanks. :smile:

**Comment by theluk** [comment]
will fix it next week

**Comment by jeremeamia** [comment]
@theluk Hey, it's been a while. I have *some* time to look at this, so if you'd like to go ahead and clean this up, I'll pull it in. Thanks.

**Comment by jeremeamia** [comment]
@theluk still interested in fixing this up. Seems like a decent addition.

**Comment by Sam-Burns** [comment]
I recommend closure in favour of https://github.com/jeremeamia/acclimate-container/pull/27 , which is still active.

**Comment by Sam-Burns** [comment]
Closed in favour of https://github.com/jeremeamia/acclimate-container/pull/27 , which is still active.

**Pull request closed without merging**